analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -44,4 +67,45 @@ protected SonarAnalysisContextBase(SonarAnalysisContext analysisContext, TContex
 
     public override bool TryGetValue<TValue>(SourceText text, SourceTextValueProvider<TValue> valueProvider, out TValue value) =>
         AnalysisContext.TryGetValue(text, valueProvider, out value);
+
+    public ProjectConfigReader ProjectConfiguration() =>
+        ProjectConfiguration(Options);
+
+    public bool IsTestProject()
+    {
+        var projectType = ProjectConfiguration(Options).ProjectType;
```

Comment:
```suggestion
        var projectType = ProjectConfiguration().ProjectType;
```

Reply:
I've noticed this and addressed the problem in #6546